baseplate/context/cassandra.py

```diff
@@ -148,8 +153,25 @@ def execute_async(self, query, parameters=None, timeout=_NOT_SET):
         future.add_errback(_on_execute_failed, span)
         return future
 
-    def prepare(self, query):
+    def prepare(self, query, cache=True):
+        """Prepares a CQL statement.
+
+        :param bool cache: If set to True (default), prepared statements will be
+        automatically cached and reused. The cache is keyed on the text of the
+        statement.
+        """
+
         trace_name = "{}.{}".format(self.context_name, "prepare")
+        query_hash = ""
+
         with self.server_span.make_child(trace_name) as span:
+            if cache:
+                query_hash = hashlib.md5(query.encode()).hexdigest()
```

Comment:
I don't think it's necessary to do hashing ourselves (or use a cryptographic hash) here. Python's dictionaries will use the built-in `hash()` on the string when dealing with lookups so it's perfectly fine to just do `cache[query.encode()]`

Reply:
TIL